codegen/masm/src/codegen/emit/int32.rs

```diff
@@ -114,8 +114,9 @@ impl<'a> OpEmitter<'a> {
     /// See `is_signed` for semantics and stack effects of the signedness check.
     #[inline]
     pub fn assert_unsigned_int32(&mut self) {
```

Comment:
I've looked into this issue, and the fix is to remove the call to `assert_unsigned_int32` from `cast` when casting from i32 to u32, from u32 to i32, and from i32 to u64; and replace it with a call to a new helper called `assert_i32`, see the implementation of `assert_i64` in `emit/int64.rs` for how to implement it.

The problem is that the `assert_unsigned_int32` helper is intended to assert that the sign bit of a 32-bit value is unset, _not_ to assert that the value is non-negative, which is how we are using it in `cast`.

All the other uses of `assert_unsigned_int32` are correct

Reply:
I have no found `assert_i32` in #242 so I made #250